### PR TITLE
Support for additonal Authentication-Schemes (digest, negotiate)

### PR DESCRIPTION
**Pull request by TheGoesen**
This change removes the "basic" Http-auth headers and replaces them with the java-Authentication API.
Advantages:  You can connect to servers which reject "basic" Authentication.
Disadvantages: Minimum java version 9, change to protected Method in HTTPTransportPipe might break existing software in the wild. 

Signed-off-by: Thorsten Goetzke th.goetzke@gmx.net

### CODE REVIEW

**Comment by TheGoesen** [comment]
turns out you need to recreate the PasswordAuthentication because Java clears the password

**Comment by TheGoesen** [comment]
I force pushed the commits using the matching e-mail syntax. Travis CI fails because the minimum java relase is now 9 instead of 8. If one simply raises the java version in the maven files the build crashes because of errors in the module system.
I am currently not planing to fix the build system. The change "works" if you compile the classes manually and update the jar.

**Comment by arjantijms** [comment]
@TheGoesen @lukasj The Java version has now been updated from 8 to the next version (11). Time to reconsider this?

**Comment by TheGoesen** [comment]
Thats a blast from the past :D 
Well I can take a look at the source code, if and how this change might be still relevant, but I no longer have access to the hardware to verify the changes

**Comment by TheGoesen** [comment]
Ok So this seems still be relevant and can probably still be adressed the same way...
So the conflict seems to be only about the imports. Can you look into this? I feat I might be a bit too lazy to redo the entire thing for a few imports...

**Comment by lukasj** [comment]
thanks!, I'll fix pending stuff after the merge...

File: jaxws-ri/runtime/rt/src/main/java/com/sun/xml/ws/transport/http/client/HttpTransportPipe.java [hunk 1]
```diff
@@ -372,19 +392,7 @@ private void recordCookies(Packet context, HttpClientTransport con) throws IOExc
...
-    private void addBasicAuth(Packet context, Map<String, List<String>> reqHeaders) {
-        String user = (String) context.invocationProperties.get(BindingProvider.USERNAME_PROPERTY);
-        if (user != null) {
-            String pw = (String) context.invocationProperties.get(BindingProvider.PASSWORD_PROPERTY);
-            if (pw != null) {
-                StringBuilder buf = new StringBuilder(user);
-                buf.append(":");
-                buf.append(pw);
-                String creds = DatatypeConverter.printBase64Binary(buf.toString().getBytes());
-                reqHeaders.put("Authorization", Collections.singletonList("Basic "+creds));
```
Comment (lukasj) [inline jaxws-ri/runtime/rt/src/main/java/com/sun/xml/ws/transport/http/client/HttpTransportPipe.java:384]: this line is important as the http request must contain appropriate header - a test failure I see on Tomcat is very likely caused by this. I'll fix it later today

**Comment by lukasj** [comment]
actually I'm to revert this for following reasons:
- basic should be still supported
- everything else is already supported by `com.sun.xml.ws.request.authenticator` property since 2.3.4 (see [JAXWSProperties](https://javadoc.io/static/com.sun.xml.ws/jaxws-rt/3.0.2/com.sun.xml.ws/com/sun/xml/ws/developer/JAXWSProperties.html#REQUEST_AUTHENTICATOR))

**Pull request merged**